api/provider/aws/load_balancer_delete.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
+	fqLoadBalancerID := addLayer0Prefix(l.Config.Instance(), loadBalancerID)
+
+	if err := l.deleteLoadBalancer(fqLoadBalancerID); err != nil {
```

Comment:
It depends on the AWS resource. For example, security groups don't have a logical name-id mapping, so we have to do a lookup to find the security group's ID before we can delete it. If it doesn't need that step, then we just go straight to deletion.